src/utils/ao.ts

```diff
@@ -158,6 +158,10 @@ export async function evolveANT({
 }
 
 export function createAoSigner(signer: ContractSigner): AoSigner {
+  if (typeof signer == 'function') {
+    return signer;
+  }
```

Comment:
nit: why not check if `signer instanceof AoSigner`?

Reply:
I thought the same, but AoSigner is a type, not a class